scalyr_agent/agent_main.py

```diff
@@ -689,12 +706,112 @@ def __handle_terminate(self):
             log.info("Received signal to shutdown, attempt to shutdown cleanly.")
             self.__run_state.stop()
 
+    def _get_system_and_agent_stats(self, status_file_path):  # type: (six.text_type) -> Optional[Dict]
+        """
+        Return current machine's and agent process' stats.
+        :return: Dict with stats.
+        """
+
+        # Get information about agent process.
+        def get_agent_process_stats():
+            agent_process_stats = {}
+            if psutil is None:
+                return {"error": "No psutil"}
+
+            pidfile = os.path.join(self.__config.agent_log_path, "agent.pid")
+
+            if not os.path.exists(pidfile):
+                return {"error": "No PID file."}
+
+            try:
+                with open(pidfile, "r") as fp:
+                    pidfile_content = fp.read().strip()
+            except OSError as e:
+                return {"error": "Error during PID file read. Error: {}".format(e)}
+
+            try:
+                psutil_process = psutil.Process(pid=int(pidfile_content))
+            except psutil.Error as e:
+                return {"error": "Can't create psutil process. Error: {}".format(e)}
+
+            try:
```

Comment:
Minor (performance) improvement - may want to use ``oneshot()`` context manager - https://psutil.readthedocs.io/en/latest/#psutil.Process.oneshot

Honestly, performance is not an issue here, but it may make sense to use oneshot in case there is a race (so we retrieve all the info / metrics for a running process at the same time - in case the process dies / goes away when we try to retrieve some other metric - probably not all that likely to happen though).